storage/mysql/mysql.go

```diff
@@ -74,31 +74,33 @@ func New(ctx context.Context, db *sql.DB, opts ...func(*tessera.StorageOptions))
 	s.queue = storage.NewQueue(ctx, opt.BatchMaxAge, opt.BatchMaxSize, s.sequenceBatch)
 
 	// Initialize checkpoint if there is no row in the Checkpoint table.
-	if _, err := s.ReadCheckpoint(ctx); err != nil {
-		if err != sql.ErrNoRows {
-			klog.Errorf("Failed to read checkpoint: %v", err)
-			return nil, err
-		}
+	if s.newCheckpoint != nil && s.parseCheckpoint != nil {
```

Comment:
This makes me quite uneasy. I think we should require these fields until we have a concrete use-case (that is tested) for doing it. At the moment `New` works if these are not present, but any attempt to integrate leaves looks like it will fail.

Reply:
From the [discussion](https://github.com/transparency-dev/trillian-tessera/pull/110#discussion_r1701644012) with @AlCutter, there is a scenario for mirrors we don't have access to the source's private key. That is why I updated the logic when I work on the test cases.

I've updated the logic to require these fields in `New`.

Reply:
Yeah I saw that conversation. My philosophy is that this precondition should be weakened at the point where we're going to add (and test!) the feature to support mirrors. Weakening the guarantee of the state now but not using it and exercising it leaves it in a more brittle state.